headers.js

```diff
@@ -7,6 +7,47 @@ var ZERO_OFFSET = '0'.charCodeAt(0)
 var USTAR = 'ustar\x0000'
 var MASK = parseInt('7777', 8)
 
+// Divide this number to get high 32 bit (JS unsafe)
+var DIVIDEND_HIGH_32BIT = Math.pow(2, 32)
+
+var MAX_SAFE_INTEGER = Number.MAX_SAFE_INTEGER || (Math.pow(2, 53) - 1)
+
+// isInteger polyfill
+// https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Number/isInteger
+var isInteger = Number.isInteger || function (value) {
+  return typeof value === 'number' && isFinite(value) && Math.floor(value) === value
+}
+
+// isSafeInteger polyfill
+// https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Number/isSafeInteger
+var isSafeInteger = Number.isSafeInteger || function (value) {
+  return isInteger(value) && Math.abs(value) <= MAX_SAFE_INTEGER
+}
+
+var getUnisgnedHigh32 = function (val) {
+  return (val / DIVIDEND_HIGH_32BIT) >>> 0
+}
+
+var getUnsignedLow32 = function (val) {
+  return val >>> 0
+}
+
+// If there's 64 bit:
+// MAX_SAFE_INT64 = (MAX_SAFE_HIGH_INT32 << 32) | MAX_SAFE_LOW_UINT32
+var MAX_SAFE_HIGH_INT32 = getUnisgnedHigh32(MAX_SAFE_INTEGER)
+var MAX_SAFE_LOW_UINT32 = getUnsignedLow32(MAX_SAFE_INTEGER)
+
+// Max individual file size
+var MAX_OCT_SIZE = parseInt('77777777777', 8)
+
+// This function is unsafe, as JavaScript can only represent 53 bit.
```

Comment:
Alternatively, use an ArrayUInt32 buffer and split the 64 bits number in two 32 bits numbers.

Reply:
Uint32arrays endianess is host dependent so this is prob better

Reply:
Should I keep as it is then...?

Reply:
Yea this looks good to me. I'll do another pass and merge later today